### PR TITLE
TW-275 [M1] What happen if user typing+ sending very fast a lot of message

### PR DESCRIPTION
**Pull request by drminh2807**
# Issue
- https://github.com/linagora/twake-on-matrix/issues/275

# Demo

https://github.com/linagora/matrix-dart-sdk/assets/12546908/7bbe3020-626e-449a-8aef-c3648cb364a9

### CODE REVIEW

File: lib/src/room.dart [hunk 1]
```diff
@@ -876,7 +875,7 @@ class Room {
           txid: messageID,
         );
       } catch (e, s) {
-        if (e is MatrixException || DateTime.now().isAfter(timeoutDate)) {
+        if ((e is MatrixException && e.error != MatrixError.M_LIMIT_EXCEEDED) || DateTime.now().isAfter(timeoutDate)) {
```
Comment (hoangdat) [inline lib/src/room.dart:878]: why not delay at exactly time back-end required?

**Review by sherlockvn — APPROVED** [review]

File: lib/src/room.dart (hunk 1 above)
Comment (drminh2807) [reply lib/src/room.dart:878]: Done!
Screenshot 2023-08-04 at 17 06 03

File: lib/src/room.dart [hunk 2]
```diff
@@ -885,8 +884,12 @@ class Room {
           _sendingQueue.remove(completer);
           return null;
         }
-        Logs().w('Problem while sending message: $e Try again in 1 seconds...');
-        await Future.delayed(Duration(seconds: 1));
+        final delay = e is MatrixException && e.raw['retry_after_ms'] is int
+            ? e.raw['retry_after_ms'] as int
+            : 1000;
```
Comment (hoangdat) [inline lib/src/room.dart:889]: indent

File: lib/src/room.dart [hunk 3]
```diff
@@ -885,8 +884,12 @@ class Room {
...
+        Logs().w(
+            'Problem while sending message: $e Try again in $delay milliseconds...');
```
Comment (hoangdat) [inline lib/src/room.dart:891]: so weird line break

**Comment by nqhhdev** [comment]
Pls squash commit

**Review by hoangdat — APPROVED** [review]

**Pull request merged**